### DISCUSSION TITLE
WireGuard support

### DISCUSSION THREAD

**Discussion by amunabv** [arendst/Tasmota#21447] [Ideas]
Would it be possible for Tasmota to support Wireguard? This would be so awesome, I'd be able to control my devices via Matter through Wireguard anywhere I go :)

I noticed someone already made an implementation for the ESP32/Arduino: https://github.com/ciniml/WireGuard-ESP32-Arduino

And there's also ESP-IDF code for Wireguard: https://github.com/trombik/esp_wireguard

ESP@Home: https://esphome.io/components/wireguard.html

**Comment by Jason2866** [comment]
If "someone" does a implementation which does not need a lot of resources there is a chance to get merged.
For secure connection there is TLS support in Tasmota. The main devs of Tasmota will not do an WireGuard implementation.

**Comment by s-hadinger** [comment]
Discussed over Discord. No recent implement to have esp32 act as a Wireguard gateway. So closing for now

**Comment by s-hadinger** [comment]
Wireguard is now available: https://github.com/arendst/Tasmota/pull/23347 and https://tasmota.github.io/docs/Wireguard/
